src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,111 @@
+import React, {
+  Dispatch,
+  ReactElement,
+  ReactNode,
+  SetStateAction,
+} from "react";
+import styled, { css, useTheme } from "styled-components";
+import { Menu } from "@headlessui/react";
+import { IconDirection } from "../Icon/Direction";
+import { Text } from "../../typography";
+import { Button } from "../Button";
+import Icon from "../Icon";
+
+export interface DropdownOption {
+  value: string;
+  label: string;
+}
+
+export interface DropdownProps {
+  buttonContent: ReactNode;
+  options: DropdownOption[];
+  selected?: DropdownOption;
+  setSelected: Dispatch<SetStateAction<DropdownOption | undefined>>;
+}
+
+export const Dropdown = ({
+  buttonContent,
+  options,
+  selected,
+  setSelected,
+}: DropdownProps): ReactElement | null => {
+  const theme = useTheme();
+
+  const isDropdownOptionSelected = ({ value }: DropdownOption): boolean => {
+    return value === selected?.value;
+  };
+
+  return (
+    <Menu>
+      {({ open }) => (
+        <>
+          <StyledMenuButton>
+            {buttonContent}
+            <ChevronComponent open={open} />
+          </StyledMenuButton>
+          <StyledMenuItems>
+            {options.map((option) => (
+              <Menu.Item key={option.value} onClick={() => setSelected(option)}>
+                {({ active }: { active: boolean }) => (
```

Comment:
Isn't headless UI typed? Do we actually need to type the active prop?

Reply:
It is, and yes, we don't need. Unsure why I've added that, I think at some point it was complaining. It would be nice to complain when there's an implicit type already.

Refactored with https://github.com/Major-Eth/yearn-web-lib/pull/28/commits/405f7df64cea9b16369cf6825a11a5f1726247d7